panic: Calling IP() on a DomainAddress. | /common/net/address.go:172

Windows
```batch
echo {"inbounds":[{"listen":"localhost","port":1,"protocol":"mixed","settings":{"udp":true}}]}|xray
```
```golang
Xray 25.7.25 (Xray, Penetrates Everything.) cfcf2a6 (go1.24.5 windows/amd64)
A unified platform for anti-censorship.
2025/07/25 23:45:13.814668 Using config from STDIN
2025/07/25 23:45:13.828149 [Info] infra/conf/serial: Reading config: &{Name:stdin: Format:json}
panic: Calling IP() on a DomainAddress.

goroutine 1 [running]:
github.com/xtls/xray-core/common/net.domainAddress.IP(...)
        github.com/xtls/xray-core/common/net/address.go:172
github.com/xtls/xray-core/transport/internet/udp.ListenUDP({0x1a30e80, 0x2431ec0}, {0x1a30f98, 0xc0002259d0}, 0x1, 0xc00033a3c0, {0xc00013db30, 0x1, 0x1a30001?})
        github.com/xtls/xray-core/transport/internet/udp/hub.go:52 +0xe9
github.com/xtls/xray-core/app/proxyman/inbound.(*udpWorker).Start(0xc00032c9c0)
        github.com/xtls/xray-core/app/proxyman/inbound/worker.go:393 +0xac
github.com/xtls/xray-core/app/proxyman/inbound.(*AlwaysOnInboundHandler).Start(0x0?)
        github.com/xtls/xray-core/app/proxyman/inbound/always.go:161 +0x45
github.com/xtls/xray-core/app/proxyman/inbound.(*Manager).Start(0xc000241cc0)
        github.com/xtls/xray-core/app/proxyman/inbound/inbound.go:121 +0x146
github.com/xtls/xray-core/core.(*Instance).Start(0xc00025acb0)
        github.com/xtls/xray-core/core/xray.go:390 +0xd2
main.executeRun(0x23d3c80?, {0xc000228630?, 0x23e9a80?, 0x700000000228620?})
        github.com/xtls/xray-core/main/run.go:90 +0x149
github.com/xtls/xray-core/main/commands/base.Execute()
        github.com/xtls/xray-core/main/commands/base/execute.go:64 +0x5ab
main.main()
        github.com/xtls/xray-core/main/main.go:22 +0x1a5
```
config.json
```json
{
 "inbounds": [
  {
   "listen": "localhost",
   "port": 1,
   "protocol": "mixed",
   "settings": {
    "udp": true
   }
  }
 ]
}
```

--- a/transport/internet/udp/hub.go
+++ b/transport/internet/udp/hub.go
@@ -40,6 +40,14 @@ func ListenUDP(ctx context.Context, address net.Address, port net.Port, streamSe
 		opt(hub)
 	}
 
+	if address.Family().IsDomain() && address.Domain() == "localhost" {
+		address = net.LocalHostIP
+	}
+
+	if address.Family().IsDomain() {
+		return nil, errors.New("domain address is not allowed for listening: ", address.Domain())
+	}
+
 	var sockopt *internet.SocketConfig
 	if streamSettings != nil {
 		sockopt = streamSettings.SocketSettings

